Clock separation

mif1-nordic opened the pull request:
Upstream PR: https://github.com/zephyrproject-rtos/zephyr/pull/104658

manifest-pr-skip

masz-nordic commented on dts/bindings/clock/nordic,nrf-clock-lfclk.yaml:
```diff
@@ -0,0 +1,55 @@
+      - "CLOCK_CONTROL_NRF_K32SRC_XTAL"
+      - "CLOCK_CONTROL_NRF_K32SRC_SYNTH"
+      - "CLOCK_CONTROL_NRF_K32SRC_EXT_LOW_SWING"
+      - "CLOCK_CONTROL_NRF_K32SRC_EXT_FULL_SWING"
```
```suggestion
      - "RC"
      - "XTAL"
      - "SYNTH"
      - "EXT_LOW_SWING"
      - "EXT_FULL_SWING"
```

masz-nordic commented on drivers/clock_control/CMakeLists.txt:
Whitespace alignment should be done in a separate PR.

masz-nordic commented on drivers/clock_control/clock_control_nrf_common.c:
```diff
@@ -0,0 +1,40 @@
+#include "clock_control_nrf_common.h"
+#include <nrfx.h>
+
+#if NRFX_CHECK(NRFX_POWER_ENABLED)
```
I'm not convinced it's a good idea to depend on nrfx config symbols.

masz-nordic commented on drivers/clock_control/clock_control_nrf_hfclk.c:
```diff
@@ -0,0 +1,426 @@
+#define ERR(dev, ...) CLOCK_LOG(ERR, dev, __VA_ARGS__)
+#define WRN(dev, ...) CLOCK_LOG(WRN, dev, __VA_ARGS__)
+#define INF(dev, ...) CLOCK_LOG(INF, dev, __VA_ARGS__)
+#define DBG(dev, ...) CLOCK_LOG(DBG, dev, __VA_ARGS__)
```
Common code between all drivers, should be factored out.

masz-nordic reviewed (commented):
Not a full review

magp-nordic reviewed (commented):
Reviewed only part of PR for now

mif1-nordic commented on drivers/clock_control/nrf_clock_calibration.c:
```diff
@@ -67,42 +73,80 @@ static int16_t prev_temperature; /* Previous temperature measurement. */
+#else
+	clk_request(DEVICE_DT_GET_ONE(COND_CODE_1((NRF_CLOCK_HAS_HFCLK),
+						  (nordic_nrf_clock_hfclk),
+						  (nordic_nrf_clock_xo))),
```
this is compatible check

mif1-nordic committed:
[nrf fromlist] drivers: clock_control: Renamed nrf2 to nrf_common.

Renamed nrf2 to nrf_common to keep the name cosistent with its use.

Upstream PR #: 104658

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

mif1-nordic committed:
[nrf fromlist] drivers: clock_control: Added common functions for separated clocks.

Added common functions for separated clocks.

Upstream PR #: 104658

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

mif1-nordic committed:
[nrf fromlist] dts: Added split clocks to dts.

Added bindings and dts nodes for split clocks.

Upstream PR #: 104658

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

mif1-nordic committed:
[nrf fromlist] drivers: clock_control: split clock control into individual clocks.

Split clock control into individual clocks:
HFCLK, XO, LFCLK, HFCLK192M, XO24M, HFCLKAUDIO

Upstream PR #: 104658

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

mif1-nordic committed:
[nrf noup] Temp fix of renamed file until upmerge

Implemented upstream fix on file that changed name in upstream.

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

mif1-nordic committed:
[nrf noup] Fix deprecation warnings for MPSL.

Fixed CONFIG_CLOCK_CONTROL_NRF deprecation warnings for the tests using MPSL.

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

pdgendt committed:
[nrf fromtree] device: Introduce DEVICE_API_EXTENDS

Some device driver APIs embed and extend other ones (e.g. I3C extends I2C).
Introduce DEVICE_API_EXTENDS to declare parent-child relationships between
device API classes, and update DEVICE_API_IS to use a new _ext_end linker
symbol as the upper bound of its range check.

All device API entries are placed in a single output section
(device_api_area). Child class entries are placed after their parent's
entries using a depth-first walk, and an _ext_end symbol is emitted after
all descendants. This makes DEVICE_API_IS(parent_class, dev) return true
for devices with a child API, and naturally supports multi-level
inheritance.

Example linker output for parent i2c with child i3c:

SECTION_PROLOGUE(device_api_area, ,) {
 Z_LINK_ITERABLE(i2c_driver_api);
 Z_LINK_ITERABLE(i3c_driver_api);
 _i3c_driver_api_ext_end = .;
 _i2c_driver_api_ext_end = .;
}

Assisted-by: Claude:claude-opus-4.6
Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

(cherry picked from commit c97a886a43bd52603b25abfa4f2472fb3977b973)

rerickson1 committed:
[nrf fromtree] doc: release-notes-4.2: dereference bl54l15u_dvk

Board was removed in favor of bl54l15_dk as a replacement.

Signed-off-by: Ryan Erickson <ryan.erickson@ezurio.com>

(cherry picked from commit 9108d1a66228c062b3dc2cd7cb34adf842a4f384)

kartben committed:
[nrf fromtree] include: drivers: clock_control: improve Doxygen documentation

Add a clock_control_interface_ext group for device-specific API
extensions and document subsystem handles and CLOCK_CONTROL_SUBSYS_ALL
with @brief tags.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

(cherry picked from commit 79900112cdb46ebaaf081a2f200d4a440b1984db)

bene42 committed:
[nrf fromtree] dts: vendor-prefixes: add eurovibes

Add Eurovibes entry

Signed-off-by: Benedikt Spranger <b.spranger@linutronix.de>

(cherry picked from commit 80460f9c1ff0f1130ac73047d9b8e5348a783b54)

pdgendt committed:
[nrf fromtree] include: drivers: nrf_clock_control: Use DEVICE_API_GET

Use helper macro to get API from struct device.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

(cherry picked from commit cf706689be670fbce3b1537719fd6d7533d449b7)

PavelVPV committed:
[nrf fromtree] samples: net: sockets: echo_client: Add support for nRF7002DK

This commit adds support for nRF7002DK in the echo_client sample.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

(cherry picked from commit c33cb2410a8506d5a4b1e56bbe96d53d1602a034)

nordicjm committed:
[nrf fromtree] boards: nordic: nrf54l15*: Do not select DT code partition for !XIP

Removes selecting a Kconfig which has no impact when XIP is
disabled as it does not build for a partition

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

(cherry picked from commit 32f5a205128af7d30893cbe345ea93005e136594)

VynDragon committed:
[nrf fromtree] dts: vendor-prefixes: Add sensylink prefix

Add sensylink prefix

Signed-off-by: Camille BAUD <mail@massdriver.space>

(cherry picked from commit 2c544a4754996752ee0e98923bd056353ecfb675)

HosakaSu committed:
[nrf fromtree] boards: jlc: add skystar_gd32f407vet6 board

Add support for the SkyStar-GD32F407VET6 board:
- Board documentation describing hardware features and usage
- Pin control for USART0 and PWM
- DTS defining GPIO, LEDs, keys, and PWM
- YAML metadata for board ID, memory, and vendor
- Default defconfig enabling MPU, console, UART, and GPIO

Signed-off-by: Liu Changjie <liucj1228@outlook.com>
Co-authored-by: RUANG (James Roy) <longjinyii@outlook.com>

(cherry picked from commit 6fb2df37bb46d6fc175964be6c4a0310fcdc4684)

rerickson1 committed:
[nrf fromtree] boards: ezurio: remove bl54L15u_dvk

This board is redundant. bl54l15_dvk is equivalent.

Signed-off-by: Ryan Erickson <ryan.erickson@ezurio.com>

(cherry picked from commit 2e1fb68e3688ddb508ae2a30c494ed773db3866c)

drew-griffin committed:
[nrf fromtree] boards: edit documentatino for FLPR core builds

In order to build `hello_world` for FLPR core SoCs, add
the `samples/hello_world` application to the west build
command.

Signed-off-by: Drew Seidel <dgriffinseidel@gmail.com>

(cherry picked from commit c8d2e7579f65a91ca637e9d52859360023656c54)

Ayush1325 committed:
[nrf fromtree] boards: beagle: beagleconnect_freedom: Add rev C5 support

There seem to be a decent number of C5 boards around. So add support for
it while keeping C7 as the default version.

The main difference between C7 and C5 is that C5 uses CC1352P instead
of CC1352P7

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

(cherry picked from commit 429c784fa8659e715035cd144e418d1203deef78)

kartben committed:
[nrf fromtree] include: drivers: clock_control: document Nordic header

Add an @file block and a clock_control_<soc> group under the
device-specific extensions group, document the public symbols, and
hide implementation details with @cond INTERNAL_HIDDEN.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>
Assisted-by: Claude:opus-4.8

(cherry picked from commit 05e4df0b7e57e4718a2c73bb0654b92d818a7ba2)

pdgendt committed:
[nrf fromtree] drivers: clock_control: nrf_clock_control: Declare API relation

Create a parent-child relation for nrf_clock_control-clock_control APIs,
this allows DEVICE_API_IS(clock_control, nrf_clock_control_dev) to return
true.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

(cherry picked from commit 6dfd1a2f8c8d16d64a2e2b390b1678e4d30d14df)

StaryAnoda committed:
[nrf fromtree] boards: remove duplicate entries in supported capability lists

Several board YAML files contained duplicate entries in their
'supported' lists. Remove the duplicates.

Affected boards:
- infineon/cyw920829m2evk_02 (cyw20829b1010, cyw20829b0lkml,
  cyw20829b1340): watchdog
- infineon/cy8cproto_062_4343w: uart
- st/nucleo_h563zi: gpio
- nxp/frdm_mcxe247: adc
- nxp/frdm_rw612: arduino_gpio
- nordic/nrf54l15dk (nrf54l10_cpuapp_ns): adc
- nordic/nrf5340dk: gpio
- raytac/an54lq_db_15: adc
- panasonic/pan1783 (evb, pa_evb, a_evb): gpio
- panasonic/panb611evb: adc
- renesas/aik_ra8d1: display, video
- gd/gd32f470i_eval: spi
- ezurio/bl54l15_dvk (nrf54l10, nrf54l15): adc
- ezurio/bl54l15u_dvk: adc
- vcc-gnd/yd_esp32: dac

Signed-off-by: Bartlomiej Fijal <StaryAnoda@Gmail.com>

(cherry picked from commit d0d232ceec1b04904690d611e638e56e7d769a3d)

rerickson1 committed:
[nrf fromtree] boards: ezurio: update bl54l15_dvk capabilities

UART30 (USB-UART port0) is now console UART.
Add mikrobus and qwiic connector support for use with shields.
Define nPM1300 PMIC on I2C bus.
Add missing nrf54l10 cpuflpr board files.

Signed-off-by: Ryan Erickson <ryan.erickson@ezurio.com>

(cherry picked from commit 83e28397a38f49d0461d9b771ecff7c7ad62b215)

chbusold committed:
[nrf fromtree] drivers: entropy: Add support for architectural entropy drivers

Add new inline function entropy_get_default_device which returns
the "zephyr,entropy" device or the architectural entropy device,
if the former is not set, and use that in all places to query the
entropy device.

This allows using architectural drivers which do not have a DT
node.

Signed-off-by: Christoph Busold <cbusold@qti.qualcomm.com>

(cherry picked from commit dad5096f854e2b15454d3d1285e80167007cbabd)

mif1-nordic committed:
[nrf fromtree] drivers: Renamed clock_control_nrf_lfclk to clock_control_nrfs_lfclk

Renamed clock_control_nrf_lfclk to clock_control_nrfs_lfclk to avoid
name clash.

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

(cherry picked from commit f9912cd4370138c4731a288248dfa1202d129cfa)

mif1-nordic committed:
[nrf fromtree] drivers: clock_control: split clock control into individual clocks.

Split clock control into individual clocks:
HFCLK, XO, LFCLK, HFCLK192M, XO24M, HFCLKAUDIO

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

(cherry picked from commit 8b92d9b807bfda56f38c1dad307dbf22d6c7724c)

kentamatsumoto1 committed:
[nrf fromtree] board: Add support for EC4L15BA1

This is a new board of the EC4L15BA1 based on nRF54L15.

Signed-off-by: Kenta Matsumoto <matsumoto_ken@jp9.kagafei.com>

(cherry picked from commit ec21c8d94a47aa4e66b5629a00a9bb60014948e5)

mif1-nordic committed:
[nrf fromtree] drivers: clock_control: Renamed nrf2 to nrf_common.

Renamed nrf2 to nrf_common to keep the name cosistent with its use.

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

(cherry picked from commit 082669e89e4544f7122415a14307a1f92ca88385)

mif1-nordic committed:
[nrf fromtree] dts: Added split clocks to dts.

Added bindings and dts nodes for split clocks.

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

(cherry picked from commit bb485e28fe774bc0a3479176f5af6f1a9a1a57ff)

mif1-nordic committed:
[nrf fromtree] drivers: clock_control: Added common functions for separated clocks.

Added common functions for separated clocks.

Signed-off-by: Michal Frankiewicz <michal.frankiewicz@nordicsemi.no>

(cherry picked from commit bca199e0bda6c336ac6fe597d10f571dbc8f2ad5)

nika-nordic committed:
[nrf fromlist] boards: nordic: nrf54l15tag: enable missing clock nodes

Enable `xo` and `lfclk` for consistency with nrf54l15dk board.

Upstream PR #: 115914

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>
